app/qml/misc/GpsDataPage.qml

```diff
@@ -27,199 +27,318 @@ Page {
 
-        Layout.fillWidth: true
 
-        titleText: qsTr( "X" )
-        text: {
-          if ( !__positionKit.hasPosition ) {
-            return qsTr( "No data from GPS" ) // if you do not have projected position yet
+            TextRowWithTitle {
+              id: gpsReceiverStatus
+
+              width: parent.width
+              height: parent.height
+
+              titleText: qsTr( "GPS receiver status" )
+              text: __positionKit.positionProvider ? __positionKit.positionProvider.statusMessage : ""
+              visible: __positionKit.positionProvider && __positionKit.positionProvider.type() === "external"
+            }
           }
-          __inputUtils.formatNumber( mapPositioning.mapPosition.x, 2 )
-        }
-      }
 
-      TextRowWithTitle {
-        id: projectY
+          TextRowWithTitle {
+            id: longitude
 
-        Layout.fillWidth: true
+            Layout.fillWidth: true
 
-        titleText: qsTr( "Y" )
-        text: {
-          if ( !__positionKit.hasPosition ) {
-            return qsTr( "No data from GPS" ) // if you do not have projected position yet
+            titleText: qsTr( "Longitude" )
+            text: {
+              if ( !__positionKit.hasPosition ) {
+                return qsTr( "No data from GPS" ) // if you do not have position yet
+              }
+              root.coordinatesInDegrees.split(", ")[0]
+            }
           }
-          __inputUtils.formatNumber( mapPositioning.mapPosition.y, 2 )
-        }
-      }
 
-      TextRowWithTitle {
-        id: horizontalAccuracy
+          TextRowWithTitle {
+            id: latitude
+
+            Layout.fillWidth: true
 
-        Layout.fillWidth: true
+            titleText: qsTr( "Latitude" )
+            text: {
+              if ( !__positionKit.hasPosition ) {
+                return qsTr( "No data from GPS" ) // if you do not have position yet
+              }
 
-        titleText: qsTr( "Horizontal accuracy" )
-        text: {
-          if ( !__positionKit.hasPosition ) {
-            return qsTr( "No data from GPS" ) // if you do not have position yet
+              let coordParts = root.coordinatesInDegrees.split(", ")
+              if ( coordParts.length > 1 )
+                return coordParts[1]
+
+              return qsTr( "No data from GPS" )
+            }
           }
 
-          __positionKit.horizontalAccuracy < 0 ? qsTr( "N/A" ) : ( __inputUtils.formatNumber( __positionKit.horizontalAccuracy, 2 ) + " m" )
-        }
-      }
+          TextRowWithTitle {
+            id: projectX
+
+            Layout.fillWidth: true
 
-      TextRowWithTitle {
-        id: verticalAccuracy
+            titleText: qsTr( "X" )
+            text: {
+              if ( !__positionKit.hasPosition ) {
+                return qsTr( "No data from GPS" ) // if you do not have projected position yet
+              }
+              __inputUtils.formatNumber( mapPositioning.mapPosition.x, 2 )
+            }
+          }
```

Comment:
does this stuff work "live" or you got it refreshed only when you close/open the panel?

Reply:
Yes, it works live, `mapPosition` is a property with `changed` signal